### PR TITLE
MetaCPAN: Add more trigger words

### LINKED ISSUE #2874: MetaCPAN: Add "mcpan" and "perl" as triggerwords
@jbarrett I tried a search for "mcpan Color::Library" and noticed it didn't trigger the IA. Figured we should add "mcpan" and "perl" as trigger words?   

------
IA Page:  http://duck.co/ia/view/meta_cpan

### PR DESCRIPTION
**Pull request by sahildua2305**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer: **`New {IA Name} Instant Answer`**
- [x] Improvement
    - [x] Bug fix: **`MetaCPAN: Fix #2874`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{SpiceRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes
Added `"perl"` and `"mcpan"` as trigger words

###### Which issues (if any) does this fix?

Fixes #2874 

###### People to notify (@mention interested parties)
@moollaza 

---

Instant Answer Page: https://duck.co/ia/view/meta_cpan

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by sahildua2305** [comment]
@moollaza I think adding `"perl"` as a trigger work will **over-trigger** this IA. What do you think?

/cc @GuiltyDolphin

**Comment by GuiltyDolphin** [comment]
@sahildua2305 Well, we can add some tests for things that shouldn't trigger, such as `perl how to use hash`.

cc @jbarrett

**Comment by sahildua2305** [comment]
@GuiltyDolphin How do you suggest adding tests for these? You mean to add check in `handle` right?

Because with `"perl"` added as a trigger word, even a query like `perl how to use hash` will trigger this IA and make an API call. :(

**Comment by GuiltyDolphin** [comment]
@sahildua2305 Ack! I forgot this was Spice (tests based on call).

@jbarrett Will we ever want to trigger if there is a multi-word input? Or are we *just* looking for package names?

**Comment by sahildua2305** [comment]
@GuiltyDolphin Marking a multi-word `remainder` will not be a good thing to do. There can be some input like `mcpan Text::Trim install`. This IA should be able to show proper results for this type of queries too.

What do you think?

**Comment by GuiltyDolphin** [comment]
@sahildua2305 @jbarrett Uhh, looks like that has its own issues:

screenshot-2016-07-04-22 21 02_743_289

Vs.

screenshot-2016-07-04-22 21 16_740_289

Looks like the relevancy went out of the window there.

**Comment by sahildua2305** [comment]
Ah! We will have to detect any such similar (like install) keyword and remove it.

File: lib/DDG/Spice/MetaCPAN.pm [hunk 1]
```diff
@@ -10,7 +10,7 @@ spice is_cached => 1;
...
 
-triggers startend => "cpan", "cpanm", "metacpan", "meta cpan", "perl library", "perl lib";
+triggers startend => "cpan", "cpanm", "metacpan", "mcpan", "meta cpan", "perl", "perl library", "perl lib";
```
Comment (jbarrett) [inline lib/DDG/Spice/MetaCPAN.pm]: I think if we add just "perl" we'll end up sending a lot of non-relevant stuff to the API - most perl queries are QA type.

We could also add "perl module", "perl dist", "perl distribution".

Comment (sahildua2305) [inline lib/DDG/Spice/MetaCPAN.pm]: Agreed that "perl" will end up over-triggering this IA. So I will remove it and add "perl module", "perl dist" and "perl distribution" instead. Thanks @jbarrett

**Comment by sahildua2305** [comment]
@jbarrett Please review this now. I have updated it. I think we can go ahead with this with hopes of improving relevancy later. What do you think?

**Comment by jbarrett** [comment]
:+1: Will hang on for @moollaza since it was his issue.

**Comment by moollaza** [comment]
Looks like we need a maintainer for this IA!

**Comment by sahildua2305** [comment]
> Looks like we need a maintainer for this IA!

I won't mind taking over, @moollaza

**Pull request merged**